packit/copr_helper.py

```diff
@@ -134,6 +110,95 @@ def create_copr_project_if_not_exists(
                 additional_packages=additional_packages,
                 additional_repos=additional_repos,
             )
+            return
+
+        delete_after_days = -1 if preserve_project else 60
+
+        fields_to_change = self.get_fields_to_change(
+            copr_proj=copr_proj,
+            additional_repos=additional_repos,
+            chroots=chroots,
+            description=description,
+            instructions=instructions,
+            list_on_homepage=list_on_homepage,
+            delete_after_days=delete_after_days,
+        )
+
+        if fields_to_change:
+            logger.info(f"Updating copr project '{owner}/{project}'")
+            for field, (old, new) in fields_to_change.items():
+                logger.debug(f"{field}: {old} -> {new}")
+
+            try:
+                self.copr_client.project_proxy.edit(
+                    ownername=owner,
+                    projectname=project,
+                    chroots=chroots,
+                    description=description,
+                    instructions=instructions,
+                    unlisted_on_hp=not list_on_homepage,
+                    additional_repos=additional_repos,
+                    delete_after_days=delete_after_days,
+                )
+            except CoprRequestException as ex:
+                if "Only owners and admins may update their projects." in str(ex):
+                    logger.info(
+                        f"We have to be admins to be able to edit project settings. "
+                        f"Requesting the admin rights for the copr '{owner}/{project}' project."
+                    )
+                    copr_proj.request_permissions(
+                        ownername=owner,
+                        projectname=project,
+                        permissions={"admin": True},
+                    )
+                raise PackitCoprSettingsException(
+                    "Copr project update failed.", fields_to_change=fields_to_change
```

Comment:
Could this log message be made more useful? Which copr project is this referring to?

Reply:
The copr repo they specified. But I can add this here as well.